test/libsolidity/semanticTests/operators/userDefined/custom_fixed_point_type_with_operators.sol

```diff
@@ -0,0 +1,22 @@
+type Fixed is int128;
+using {add as +, mul as *} for Fixed;
+
+int constant MULTIPLIER = 10**18;
+
+function add(Fixed a, Fixed b) pure returns (Fixed) {
+    return Fixed.wrap(Fixed.unwrap(a) + Fixed.unwrap(b));
+}
+
+function mul(Fixed a, Fixed b) pure returns (Fixed) {
+    int intermediate = (int(Fixed.unwrap(a)) * int(Fixed.unwrap(b))) / MULTIPLIER;
+    if (int128(intermediate) != intermediate) { revert("Overflow"); }
+    return Fixed.wrap(int128(intermediate));
+}
+
+contract C {
+    function applyInterest(Fixed value, Fixed percentage) public pure returns (Fixed result) {
+        return value + value * percentage;
```

Comment:
Attaching a function to an operator will be evaluated in the order of precedence of the operator, right? I.e. `1 + 2 * 3` will be `7` and not `9`. I've not yet gone over all of the tests, but it would be nice if we had a more trivial example of this in action in the form of a test, i.e. just some simple arithmetic.

Reply:
Done. Added `calling_all_operators_precedence.sol` that makes sure that the precedence matches the built-in operators.